kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -1361,6 +1373,69 @@ impl<'tcx> GotocCtx<'tcx> {
         )
     }
 
+    /// Generates code for a SIMD vector comparison intrinsic.
+    ///
+    /// We perform some typechecks here for two reasons:
+    ///  * In the case of SIMD intrinsics, these checks depend on the backend.
+    ///  * We can emit a friendly error here, but not in `cprover_bindings`.
+    fn codegen_intrinsic_simd_cmp<F: FnOnce(Expr, Expr, Type) -> Expr>(
+        &mut self,
+        f: F,
+        mut fargs: Vec<Expr>,
+        p: &Place<'tcx>,
+        span: Option<Span>,
+        ret_typ: Type,
+    ) -> Stmt {
+        let arg1 = fargs.remove(0);
+        let arg2 = fargs.remove(0);
+
+        // The return type must be the same length as the input types. The
```

Comment:
nit: I personally prefer expanding the method documentation and keeping the inline documentation small. But it's up to you.

My reasoning is:
1. It's clear in the method signature what it checks.
2. It makes it easier to read the code.
3. Separation of concerns. Method signature documentation talks about the what and inline code comments talk about the how.

Reply:
Thanks for the explanation, it made a lot of sense to me. Moved these comments to the method documentation!